src/vmm/src/vmm_config/net.rs

```diff
@@ -310,44 +292,23 @@ mod tests {
         // Error Cases for UPDATE
         // Error Case: Update netif_2 mac using the same mac as netif_1.
         let netif_2 = create_netif(id_2, host_dev_name_2, guest_mac_1);
-        let expected_error = format!("The guest MAC address {} is already in use.", guest_mac_1);
+        let expected_error = NetworkInterfaceError::GuestMacAddressInUse(guest_mac_1.into());
         assert_eq!(
             net_builder.build(netif_2).err().unwrap().to_string(),
-            expected_error
+            expected_error.to_string()
         );
 
         // Error Case: Update netif_2 dev_host_name using the same dev_host_name as netif_1.
         let netif_2 = create_netif(id_2, host_dev_name_1, guest_mac_2);
         assert_eq!(
             net_builder.build(netif_2).err().unwrap().to_string(),
             NetworkInterfaceError::CreateNetworkDevice(devices::virtio::net::Error::TapOpen(
-                TapError::IoctlError(std::io::Error::from_raw_os_error(16))
+                TapError::IfreqExecuteError(std::io::Error::from_raw_os_error(16))
             ))
             .to_string()
         );
     }
 
-    #[test]
-    fn test_error_display() {
```

Comment:
This test only tests if `Display` and `Debug` traits are implemented for the error type. So basically the only way this test would fail is when these traits are not implemented for the error, but the test would never run in this case, because it would not compile. So the test does not `test` for anything during runtime and will always be successful. In my opinion such test is worthless.

Reply:
yes, I agree with @ShadowCurse that these type of tests are kind of superfluous.

Reply:
I could have clarified my point here better.

Given there are new error types being introduced, I would have expected some test code to make use of them. If you have a concern with the completeness of the tests that is completely fine. I maintain that it would be a bit better to extend rather than just delete.

With that said, testing that Display and Debug have been implemented is not a bad thing, but it isn't the point that I think is more important around addressing the coverage.

Reply:
Well, you can say that there are new error variants, but in reality I just split one generic error into several more specific. Actually there are tests for `SetOffloadFlags` and `SetSizeOfVnetHdr` errors [here](https://github.com/firecracker-microvm/firecracker/blob/3dd605ad678613294369c492b3966a469239a758/src/devices/src/virtio/net/tap.rs#L250) for example, they are just not easily visible (and I never touched them). 
So I don't see how to expand testing on this errors, because tests for them already existed. 
In my opinion it is better to have a lower coverage with actually useful tests, rather then high coverage with `Debug`, `Display` tests.

Reply:
Can you confirm which tests make use of the new error types?